client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamFactory.java

```diff
@@ -25,9 +25,10 @@
      * @param txId    The transaction id.
      * @param segmentSealedCallback Method to be executed on receiving SegmentSealed from SSS.
      * @param config  The configuration for the writer
+     * @param delegationToken token to pass on to segmentstore to authenticate access to the segment.
```

Comment:
Please find a way to encapsulate the delegation token within the impl.

Reply:
1. I think you mean encapsulate this in `SegmentOutputStreamFactoryImpl`. The lifetime for the token and the impl are different. 
2. The life time of `SegmentOutputStreamImpl` object and the delegation token is same (we destroy the outputstream in case of auth failure). Whereas `delegationtoken` is retrieved from the latest call to controller in the context of this resource. By creating it within this object we will incur one extra roundtrip to controller.

Reply:
I was thinking you could encapsulate everything in the `ClientFactoryImpl`.